File: Data Structures & Algorithms/jump-game-ii/submission-1.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size()-1;

        int jumpCount = 0;
        int currJump = 0;
        int farthest = 0;

        for(int i = 0; i < n; i++)
        {
            farthest = max(farthest, i + nums[i]);
            if(i == currJump){
                jumpCount++;
                currJump = farthest;
            }
        }
        return jumpCount;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = nums[0]

        resMaxSubArray = nums[0]

        for i in range(1,  len(nums)):
            resMaxSubArray = max(resMaxSubArray + nums[i], nums[i])

            res = max(res, resMaxSubArray)
        return res